xarray/core/variable.py

```diff
@@ -1978,8 +1979,14 @@ def no_conflicts(self, other, equiv=duck_array_ops.array_notnull_equiv):
         return self.broadcast_equals(other, equiv=equiv)
 
     def quantile(
-        self, q, dim=None, interpolation="linear", keep_attrs=None, skipna=True
-    ):
+        self,
+        q: np.typing.ArrayLike,
+        dim=None,
```

Comment:
I cannot type check `dim` without changing behaviour for `quantile(q, dim=("x", "y"))`.